cmd/nginx-ingress/aws.go

```diff
@@ -0,0 +1,108 @@
+// +build aws
+
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"log"
+	"math/rand"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/marketplacemetering"
+	"github.com/aws/aws-sdk-go-v2/service/marketplacemetering/types"
+
+	"github.com/dgrijalva/jwt-go/v4"
+)
+
+const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-"
+
+var (
+	productCode   string
+	pubKeyVersion int32 = 1
+	pubKeyString  string
+	nonce         string
+)
+
+func init() {
+	rand.Seed(jwt.Now().UnixNano())
+	nonce = RandStringBytesRmndr(32)
+
+	cfg, err := config.LoadDefaultConfig(context.TODO())
```

Comment:
It's probably worth implement the context here correctly. https://blog.golang.org/context

Reply:
Which context should we use? In the docs and examples they always use `context.TODO()` when loading the config.
https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/#loading-aws-shared-configuration

Reply:
can it make a network call? If that is case, it makes sense to add a `context.WithTimeout...` so that we control the timeout.

Reply:
Your right, I looked at the docs and its all `context.TODO()` 

As Michael said, it's usually a timeout for network calls.

Reply:
`LoadDefaultConfig` could make a network call, I think it's the last resource if it can't find a valid config. `RegisterUsage` of course makes a network call, I don't think I see the value of controlling the timeout, but I've added it. It's set to 5 seconds now, no idea what an acceptable timeout is.